cpp/src/reductions/compound.cuh

```diff
@@ -54,35 +54,36 @@ std::unique_ptr<scalar> compound_reduction(column_view const& col,
 {
   auto const valid_count = col.size() - col.null_count();
 
+  // ddof is allowed to be negative, hence need to check for all nulls
```

Comment:
*question*: pandas at least doesn't raise for negative ddof, but it is mathematically nonsensical, so arguably libcudf should `CUDF_EXPECTS(ddof >= 0, ...)` in the public API.

Reply:
Did this.